include/thread_pool/worker.hpp

```diff
@@ -349,13 +366,14 @@ inline void Worker<Task, Queue>::threadFunc(size_t id, WorkerVector& workers, Sl
             for (auto i = 0u; i < m_busy_wait_options.numIterations() && !task_found; i++)
             {
                 std::this_thread::sleep_for(m_busy_wait_options.iterationFunction()(i));
-                task_found = tryHandleTask(handler, workers);
+                task_found = tryGetTask(handler, workers);
             }
 
             // If we found a task during our busy wait sequence, we abort it and transition back into the active loop.
             if (task_found)
             {
                 num_busy_waiters.fetch_sub(1, std::memory_order_acq_rel);
+                handleTask(handler); // Handle the task body only once we decrement the busy waiting loop.
```

Comment:
I split `tryHandleTask` into `tryGetTask` and `handleTask`. This allows us to handle the task _after_ exiting the busy wait state.